async/apply.js

```diff
@@ -1,24 +1,52 @@
+const pull = require('pull-stream')
+const { heads } = require('ssb-sort')
+const { isFeed } = require('ssb-ref') // TODO check this is name of method!
 const isTag = require('../sync/isTag')
 
-module.exports = function(server, api) {
-  return function(data, cb) {
-    const { tagged, message, recps, tag } = data
-    const msg = recps && recps.length > 0 ? {
-      type: 'tag',
-      tagged,
-      message,
-      root: tag,
-      branch: tag,
-      recps: data.recps,
-      private: true
-    } : {
-      type: 'tag',
-      tagged,
-      message,
-      root: tag,
-      branch: tag
+module.exports = function (server) {
+  return function applyTag (data, cb) {
+    const { tagged, message, recps = [], tag } = data
+    const _recps = recps.filter(feed => {
+      return isFeed(feed) || isFeed(feed.link) // TODO check this logic
+    })
+
+    pull(
+      createBacklinkStream(tag),
+      pull.collect((err, msgs) => {
+        if (err) return cb(err)
+
+        var msg = {
+          type: 'tag',
+          tagged,
+          message,
+          root: tag,
+          branch: heads(msgs)
```

Comment:
I assume if a tag was used widely this list would grow quite large, potentially too large. Should there be any mitigation of that scenario?

Reply:
I can explain how heads works on the call. Basically no, this would be 1-2 message ids in general